src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -6996,6 +6998,31 @@ static void sofia_handle_sip_r_options(switch_core_session_t *session, int statu
 	}
 }
 
+/* Check if we should skip directory lookup for 3xx redirects */
+static switch_bool_t sofia_should_skip_directory_lookup(sofia_profile_t *profile, int status, const char *host)
+{
+	char domains[1024], *list[64];
+	int i, n;
+	
+	/* Handle all redirect status codes: 300, 301, 302, 305 */
+	if ((status != 300 && status != 301 && status != 302 && status != 305) 
+		|| zstr(profile->redirect_no_lookup_domains) || zstr(host))
+		return SWITCH_FALSE;
+	
+	switch_copy_string(domains, profile->redirect_no_lookup_domains, sizeof(domains));
+	n = switch_separate_string(domains, ',', list, 64);
```

Comment:
Minor change, since we are always gonna do this, I think we can perform tokenize when reading the configuration. So in here its just gonna loop the list of allowed domains